src/main.js

```diff
@@ -1,29 +1,51 @@
 //function that checks if the env value is valid
 //value cannot start with a digit and must only contain alphanumeric characters or an underscore
 const checkValidationOfValues = envSpecString => {
+  const validValues = ['color','date','datetime-local','email','month','number','password','tel','text','time','url','week','default'];
   const alphanumericThatDoesNotStartWithDigit = /^[A-Z_][0-9A-Z_]*$/;
-  let envValuesChecked = envSpecString.split("\n"); //split lines based on \n character
+  let envValArray = parseVarFromVal(envSpecString.split("\n")); //split lines based on \n character and parse them
 
-  return (envValuesChecked = envValuesChecked.filter(element => {
-    return element.match(alphanumericThatDoesNotStartWithDigit); //keep valid values
-  }));
+  return envValArray = envValArray.filter( element =>{ //check for valid variables AND values
+    if(element[0].match(alphanumericThatDoesNotStartWithDigit)&&(validValues.includes(element[1]))){
+      return element;
+    }
+  })
 };
 
+//function that separates the variables from their values and returns them as separated arrays
+const parseVarFromVal = envSpecAsArray => {
+  let envArrayParsed = [];
+  let parsPos = -1;
+  return envArrayParsed = envSpecAsArray.map( element =>{
+    parsPos = element.indexOf(":");
+    if(parsPos<0){
+      parsPos=element.length
+      return [element.substring(0,parsPos).trim(),'default'];
+    }
+    else{
+      return [element.substring(0,parsPos).trim(),element.substring(parsPos+1,).trim()];
+    }
+  })
+};
+
+
 //function that returns the HTML code as a string
 const outputHTML = envValues => {
   //create HTML format
   envValues = envValues.map(
     element =>
-      `<label for="env_spec_${element.toLowerCase()}">${element}</label>\n` +
-      `<input id="env_spec_${element.toLowerCase()}" name="${element.toLowerCase()}" />\n`
+      `<label for="env_spec_${element[0].toLowerCase()}">${element[0]}</label>\n` +
+      `<input id="env_spec_${element[0].toLowerCase()}" name="${element[0].toLowerCase()}" />\n`
```

Comment:
We should render the `input`'s type here as described in the example here: https://github.com/sourcelair/env-spec-js/issues/3.